docs: update docs to reflect changes in Angular CLI version 17.1

alan-agius4 opened the pull request:
This changes include mentioning the experimental application builder migration and changes to the `outputPath` and `optimization` option.

jnizet commented:
Oops, wrong PR approved. Sorry for the noise.

dgp1130 commented on adev/src/content/reference/configs/workspace-config.md:
```diff
@@ -508,3 +509,16 @@ When supplying the value as a string the filename of the specified path will be
+| `base`    | Specify the output path relative to workspace root.                                | `string`   |               |
+| `browser` | The output directory name of your browser build within the output path base.       | `string`   | `browser`     |
+| `server`  | The output directory name of your server build within the output path base.        | `string`   | `server`      |
+| `media`   | The output directory name of your media files within the output browser directory. | `string`   | `media`       |
```
Consider: Expand on the meaning of "browser", "server", and "media". We can maybe call out that `browser` is safe to serve to public users. Exactly what content goes into `media` might be good to clarify as well.

Don't we also have a constraint that the directories can't nest inside each other? Anything we want to state there?

dgp1130 commented on adev/src/content/tools/cli/esbuild.md:
```diff
@@ -117,25 +117,13 @@ The `application` builder now provides the integrated functionality for all of t
-    This does not apply to third-party dependencies.
-
-In the future, a schematic will make this migration process easier for existing applications.
+To convert from the separate SSR builders to the integrated capabilities of the `application` builder you are required to run an experimental migration.
```
Active voice:

> To convert from the separate SSR builders to the integrated capabilities of the `application` builder, run the experimental `use-application-builder` migration.

dgp1130 commented on adev/src/content/tools/cli/esbuild.md:
```diff
@@ -117,25 +117,13 @@ The `application` builder now provides the integrated functionality for all of t
-    This does not apply to third-party dependencies.
-
-In the future, a schematic will make this migration process easier for existing applications.
+To convert from the separate SSR builders to the integrated capabilities of the `application` builder you are required to run an experimental migration.
```
Question: Regarding "experimental", are we calling this schematic experimental because it is optional or is there another milestone we're aiming for to call it "stable"?

I'm fine with experimental for now, but I just want to make sure I'm understanding the intent behind this and to make sure we don't forget to remove "experimental" at the appropriate time.

alan-agius4 replied:
It is actually experimental as we decided that whilst this is optional this is not listed as part of the optional migrations for the time being.

dgp1130 commented on adev/src/content/tools/cli/esbuild.md:
```diff
@@ -117,25 +117,13 @@ The `application` builder now provides the integrated functionality for all of t
+
+ng update @angular/cli --name use-application-builder
+
+</docs-code>
```
Suggestion: Expand on what this does either before or after the code block. Doesn't need to be as expansive as before, maybe something along the lines of:

* Converts existing `browser` or `browser-esbuild` target to `application`.
* Removes any previous SSR builders (because `application` does that now).
* Updates configuration accordingly.
* Removes Webpack specific elements.

I think this would help users understand what this schematic is actually doing.

dgp1130 commented on adev/src/content/tools/cli/esbuild.md:
```diff
@@ -117,25 +117,13 @@ The `application` builder now provides the integrated functionality for all of t
-    An example of the changes for a v16 project that has been converted can be found [here](https://github.com/alan-agius4/angular-cli-use-application-builder/commit/1defdb93a7f508662bc427439e51505668bf84cd#diff-1ba718c1eb8aa39cd20c2562d92523068c734d75f54655e97d652b992d9b4259).
-6.  Remove any CommonJS assumptions in the application server code such as `require`, `__filename`, `__dirname`, or other constructs from the [CommonJS module scope](https://nodejs.org/api/modules.html#the-module-scope).
-    All application code should be ESM compatible.
-    This does not apply to third-party dependencies.
```
Question: Do we still want to mention any constraints or patterns that the schematic can't handle such as `__dirname` or `require`? Does the schematic detect those patterns or will things just fail afterwards? Anything we want to say in that respect?

dgp1130 commented on adev/src/content/tools/cli/esbuild.md:
```diff
@@ -175,8 +163,6 @@ Several build options are not yet implemented but will be added in the future as
 
 - [WASM imports](https://github.com/angular/angular-cli/issues/25102) -- WASM can still be loaded manually via [standard web APIs](https://developer.mozilla.org/en-US/docs/WebAssembly/Loading_and_running).
 
-Building libraries with the new build system via `ng-packagr` is also not yet possible but library build support will be available in a future release.
```
Question: This line still seems accurate, and I can expect users to ask that question. Any particular reason you want to remove it here?

alan-agius4 replied:
Maybe I am not understanding the statement but  consuming libraries that are built with ng-packagr works.

dgp1130 replied:
Oh interesting, my interpretation was that this was talking about _building_ libraries directly. As in, there's no current way to `ng generate library` and then build it with ESBuild.

swami-sanapathi commented on adev/src/content/tools/cli/esbuild.md:
```diff
@@ -117,25 +117,13 @@ The `application` builder now provides the integrated functionality for all of t
-    This does not apply to third-party dependencies.
-
-In the future, a schematic will make this migration process easier for existing applications.
+To convert from the separate SSR builders to the integrated capabilities of the `application` builder, run the experimental `use-application-builder migration.
```
To convert from the separate SSR builders to the integrated capabilities of the `application` builder, run the experimental `use-application-builder` migration.

swami-sanapathi commented on aio/content/guide/esbuild.md:
```diff
@@ -124,26 +124,13 @@ The `application` builder now provides the integrated functionality for all of t
-This does not apply to third-party dependencies.
-
-In the future, a schematic will make this migration process easier for existing applications.
+To convert from the separate SSR builders to the integrated capabilities of the `application` builder, run the experimental `use-application-builder migration.
```
To convert from the separate SSR builders to the integrated capabilities of the `application` builder, run the experimental `use-application-builder` migration.

alan-agius4 committed:
docs: update docs to reflect changes in Angular CLI version 17.1

This changes include mentioning the experimental application builder migration and changes to the `outputPath` and `optimization` option.

alan-agius4 committed:
fixup! docs: update docs to reflect changes in Angular CLI version 17.1

pkozlowski-opensource commented:
This PR was merged into the repository by commit a3787bdfa61ef98c9e109d62c49e7bf873c03807.
